lib/ansible/module_utils/csharp/Ansible.Basic.cs

```diff
@@ -248,6 +250,33 @@ public void Deprecate(string message, string version)
             LogEvent(String.Format("[DEPRECATION WARNING] {0} {1}", message, version));
         }
 
+        public void Deprecate(string message, string version, string collectionName)
+        {
+            // `collectionName` is a Ansible 2.10 parameter. We accept and ignore it,
+            // to avoid modules/plugins from 2.10 conformant collections to break with
+            // new enough versions of Ansible 2.9.
+            deprecations.Add(new Dictionary<string, string>() { { "msg", message }, { "version", version } });
+            LogEvent(String.Format("[DEPRECATION WARNING] {0} {1}", message, version));
+        }
+
+        public void Deprecate(string message, DateTime date)
+        {
+            // This function is only available for Ansible 2.10. We still accept and ignore it,
+            // to avoid modules/plugins from 2.10 conformant collections to break with new enough
+            // versions of Ansible 2.9.
+            deprecations.Add(new Dictionary<string, string>() { { "msg", message }, { "version", null } });
+            LogEvent(String.Format("[DEPRECATION WARNING] {0} {1}", message, null));
+        }
+
+        public void Deprecate(string message, DateTime date, string collectionName)
+        {
+            // This function is only available for Ansible 2.10. We still accept and ignore it,
+            // to avoid modules/plugins from 2.10 conformant collections to break with new enough
+            // versions of Ansible 2.9.
+            deprecations.Add(new Dictionary<string, string>() { { "msg", message }, { "version", null } });
+            LogEvent(String.Format("[DEPRECATION WARNING] {0} {1}", message, null));
+        }
```

Comment:
Can avoid having this defined twice with a similar body

```suggestion
        public void Deprecate(string message, DateTime date, string collectionName = null)
        {
            // This function is only available for Ansible 2.10. We still accept and ignore it,
            // to avoid modules/plugins from 2.10 conformant collections to break with new enough
            // versions of Ansible 2.9.
            deprecations.Add(new Dictionary<string, string>() { { "msg", message }, { "version", null } });
            LogEvent(String.Format("[DEPRECATION WARNING] {0} {1}", message, null));
        }
```

Reply:
Fixed as discussed on IRC.